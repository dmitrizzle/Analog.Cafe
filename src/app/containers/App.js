// tools
import React from "react"
import { Router, Route, browserHistory, IndexRoute } from "react-router"
import Helmet from "react-helmet"
import ReactGA from "react-ga"

// theme
import Paper from "../themes/Paper"


// views and components
import { About, NotFound, Login, Publication } from "../components/Views"
import { Intro, Submit } from "../components/Views/Submit"
import Composer from "./Composer"
import List from "./List"
import Post from "./Post"


// init GA tracking
ReactGA.initialize("UA-91374353-3")

// render & route
export default props => {
	const updateView = () => {
		ReactGA.set({ page: window.location.pathname + window.location.search })
		ReactGA.pageview(window.location.pathname + window.location.search)

		window.scrollTo(0,0)
	}
	return (
		<Paper>
			<Helmet
				defaultTitle="Analog.Cafe ☕️"
				titleTemplate="%s ☕️ Analog.Cafe"
			/>

			<Router history={ browserHistory } onUpdate={ updateView } >

				<Route path="/"			 					component={ Publication } >
					<IndexRoute 								component={ List } />
					<Route path="photo-essays"	component={ List } />
					<Route path="articles"			component={ List } />
					<Route path="about"			 		component={ About } />

					<Route path="zine/*"				component={ Post } />
					<Route path="author/*"			component={ List } />
				</Route>

				<Route path="submit"					component={ Submit } >
					<IndexRoute 								component={ Intro } />
					<Route path="compose" 			component={ Composer } />
				</Route>

				<Route path="login"						component={ Login } />
				<Route path="*"								component={ NotFound } status={404} />

			</Router>

		</Paper>
	)
}
