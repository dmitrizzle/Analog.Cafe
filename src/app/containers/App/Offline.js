// tools
import React from "react"
import Loadable from "react-loadable"
import Link from "../../components/_controls/Link"

// redux
import { connect } from "react-redux"
import { addAvailableComponent } from "../../../actions/userActions"

class EnableOffline extends React.PureComponent {
  handleClick = event => {
    event.preventDefault()
    Loadable.preloadAll()
      .then(() => {
        this.props.addAvailableComponent("Composer")
        this.forceUpdate()
      })
      .catch(error => console.log(error))
  }
  render = () => {
    if (
      this.props.user.connection.status !== "offline" ||
      this.props.user.connection.availableComponents.indexOf("Composer") !== -1
    )
      return (
        <small style={{ opacity: "0.5" }}>
          {this.props.user.connection.availableComponents.indexOf(
            "Composer"
          ) === -1 ? (
            <span>
              <Link to="#download" onClick={this.handleClick}>
                Enable
              </Link>{" "}
              offline editing.
            </span>
          ) : (
            <span>You can now edit while offline.</span>
          )}
        </small>
      )
    else return null
  }
}

// connect with redux
const mapStateToProps = state => {
  return {
    user: state.user
  }
}
const mapDispatchToProps = dispatch => {
  return {
    addAvailableComponent: componentName => {
      dispatch(addAvailableComponent(componentName))
    }
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(EnableOffline)
