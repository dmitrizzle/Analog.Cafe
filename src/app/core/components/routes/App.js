import { Redirect, Route, Switch } from "react-router-dom"
import Loadable from "react-loadable"
import React from "react"

import { ROUTE_URL_USER_LANDING } from "../../../user/constants/routes-session"
import ArticleLoader from "../pages/Article/components/ArticleLoader"
import ListLoader from "../pages/List/components/ListLoader"
import PrivacySettings from "../../../user/components/pages/PrivacySettings"
import SignOut from "../../../user/components/pages/SignOut"
import Subscribe from "../pages/Subscribe"

const List = Loadable({
  loader: () => import("../pages/List"),
  loading: ListLoader
})
const Article = Loadable({
  loader: () => import("../pages/Article"),
  loading: ArticleLoader,
  delay: 100
})
const EditProfile = Loadable({
  loader: () => import("../../../user/components/pages/EditProfile"),
  loading: ArticleLoader,
  delay: 100
})
const About = Loadable({
  loader: () => import("../pages/About"),
  loading: ArticleLoader,
  delay: 100
})
const SignIn = Loadable({
  loader: () => import("../../../user/components/pages/SignIn"),
  loading: ArticleLoader,
  delay: 100
})
const Me = Loadable({
  loader: () => import("../../../user/components/pages/Me"),
  loading: ArticleLoader,
  delay: 100
})
const Admin = Loadable({
  loader: () => import("../../../admin/components/pages/Admin"),
  loading: ArticleLoader,
  delay: 100
})
const Submit = Loadable({
  loader: () => import("../../../user/components/routes/Submit"),
  loading: ArticleLoader,
  delay: 100
})
const NotFound = Loadable({
  loader: () => import("../pages/Error/components/NotFound"),
  loading: ArticleLoader,
  delay: 100
})

const PrivacyPolicy = Loadable({
  loader: () => import("../../../user/components/pages/PrivacyPolicy"),
  loading: ArticleLoader,
  delay: 100
})

export default () => {
  return (
    <main
      style={{
        transform: "translateZ(0)",
        marginTop: "-8em",
        paddingTop: "8em"
      }}
    >
      <Switch>
        {/* dynamic urls and views */}
        <Route exact path="/author" component={NotFound} />
        <Route exact path="/zine" component={NotFound} />
        <Route exact path="/author/*" component={List} />
        <Route exact path="/zine/*" component={Article} />
        {/* NOTE below is a set of outdated routes */}
        <Route exact path="/articles" render={() => <Redirect to="/" />} />
        <Route
          exact
          path="/stories"
          render={() => <Redirect to="/photo-essays" />}
        />
        <Route
          exact
          path="/storys"
          render={() => <Redirect to="/photo-essays" />}
        />
        <Route
          exact
          path="/guides"
          render={() => <Redirect to="/film-photography" />}
        />
        <Route
          exact
          path="/reviews"
          render={() => <Redirect to="/film-photography" />}
        />
        <Route
          exact
          path="/film-cameras"
          render={() => <Redirect to="/film-photography" />}
        />
        <Route
          exact
          path="/places"
          render={() => <Redirect to="/photo-essays" />}
        />
        <Route
          exact
          path="/art"
          render={() => <Redirect to="/photo-essays" />}
        />
        <Route
          exact
          path="/photo-stories"
          render={() => <Redirect to="/photo-essays" />}
        />
        {/* new magazine sections */}
        <Route exact path="/" component={List} />
        <Route exact path="/film-photography" component={List} />
        <Route exact path="/photo-essays" component={List} />
        <Route exact path="/editorials" component={List} />
        <Route exact path="/unclassifieds" component={List} />
        <Route exact path="/collaborations" component={List} />
        <Route exact path="/solo-projects" component={List} />
        {/* auth views */}
        <Route exact path={ROUTE_URL_USER_LANDING} component={Me} />
        <Route
          exact
          path={`${ROUTE_URL_USER_LANDING}/admin`}
          component={Admin}
        />
        <Route exact path="/submissions/*" component={Article} />
        <Route
          exact
          path={ROUTE_URL_USER_LANDING + "/edit"}
          component={EditProfile}
        />
        <Route exact path="/sign-in" component={SignIn} />
        <Route exact path="/sign-out" component={SignOut} />
        {/* static views and urls */}
        <Route exact path="/about" component={About} />
        <Route path="/submit" component={Submit} />
        <Route path="/subscribe" component={Subscribe} />
        <Route exact path="/privacy-policy" component={PrivacyPolicy} />
        <Route exact path="/privacy-settings" component={PrivacySettings} />
        <Route state={{ status: "404" }} component={NotFound} />
      </Switch>
    </main>
  )
}
