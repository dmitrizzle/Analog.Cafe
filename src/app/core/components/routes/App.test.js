import "jest-styled-components"

import { shallow } from "enzyme"
import React from "react"

import App from "./App.js"

it("Render routes App, matches snapshot", () => {
  const element = shallow(<App />)
  expect(element).toMatchSnapshot()
})
