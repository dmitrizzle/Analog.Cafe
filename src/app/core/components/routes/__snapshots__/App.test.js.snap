// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render routes App, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
                component={[Function]}
                exact={true}
                path="/author"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/zine"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/author/*"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/zine/*"
        />
        <Route
                exact={true}
                path="/photo-essays"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/articles"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/stories"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/storys"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/guides"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/reviews"
                render={[Function]}
        />
        <Route
                component={[Function]}
                exact={true}
                path="/"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/film-cameras"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/places"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/art"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/photo-stories"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/editorials"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/unclassifieds"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/collaborations"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/solo-projects"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/me"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/me/admin"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/submissions/*"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/me/edit"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/sign-in"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/sign-out"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/about"
        />
        <Route
                component={[Function]}
                path="/submit"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/privacy-policy"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/privacy-settings"
        />
        <Route
                component={[Function]}
                state={
                        Object {
                                "status": "404",
                              }
                }
        />
</Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/author"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/zine"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/author/*"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/zine/*"
/>,
          <Route
            exact={true}
            path="/photo-essays"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/articles"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/stories"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/storys"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/guides"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/reviews"
            render={[Function]}
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/film-cameras"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/places"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/art"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/photo-stories"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/editorials"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/unclassifieds"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/collaborations"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/solo-projects"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/me"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/me/admin"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/submissions/*"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/me/edit"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/sign-in"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/sign-out"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/about"
/>,
          <Route
            component={[Function]}
            path="/submit"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/privacy-policy"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/privacy-settings"
/>,
          <Route
            component={[Function]}
            state={
                        Object {
                                    "status": "404",
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/author",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/zine",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/author/*",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/zine/*",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/photo-essays",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/articles",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/stories",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/storys",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/guides",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/reviews",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/film-cameras",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/places",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/art",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/photo-stories",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/editorials",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/unclassifieds",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/collaborations",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/solo-projects",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/me",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/me/admin",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/submissions/*",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/me/edit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/sign-in",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/sign-out",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/about",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/submit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/privacy-policy",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/privacy-settings",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "state": Object {
              "status": "404",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/author"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/zine"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/author/*"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/zine/*"
          />
          <Route
                    exact={true}
                    path="/photo-essays"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/articles"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/stories"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/storys"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/guides"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/reviews"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/film-cameras"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/places"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/art"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/photo-stories"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/editorials"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/unclassifieds"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/collaborations"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/solo-projects"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/me"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/me/admin"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/submissions/*"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/me/edit"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/sign-in"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/sign-out"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/about"
          />
          <Route
                    component={[Function]}
                    path="/submit"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/privacy-policy"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/privacy-settings"
          />
          <Route
                    component={[Function]}
                    state={
                              Object {
                                        "status": "404",
                                      }
                    }
          />
</Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/author"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/zine"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/author/*"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/zine/*"
/>,
            <Route
              exact={true}
              path="/photo-essays"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/articles"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/stories"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/storys"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/guides"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/reviews"
              render={[Function]}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/film-cameras"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/places"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/art"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/photo-stories"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/editorials"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/unclassifieds"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/collaborations"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/solo-projects"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/me"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/me/admin"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/submissions/*"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/me/edit"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/sign-in"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/sign-out"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/about"
/>,
            <Route
              component={[Function]}
              path="/submit"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/privacy-policy"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/privacy-settings"
/>,
            <Route
              component={[Function]}
              state={
                            Object {
                                          "status": "404",
                                        }
              }
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/author",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/zine",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/author/*",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/zine/*",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/photo-essays",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/articles",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/stories",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/storys",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/guides",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/reviews",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/film-cameras",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/places",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/art",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/photo-stories",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/editorials",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/unclassifieds",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/collaborations",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/solo-projects",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/me",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/me/admin",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/submissions/*",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/me/edit",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/sign-in",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/sign-out",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/about",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/submit",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/privacy-policy",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/privacy-settings",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "state": Object {
                "status": "404",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
